ogr/utils.py

```diff
@@ -234,7 +151,4 @@ def filter_paths(paths: List[str], filter_regex: str) -> List[str]:
     :return: [str]
     """
     pattern = re.compile(filter_regex)
-    paths = list(
-        filter(lambda path: (not pattern or bool(pattern.search(path))), paths)
-    )
-    return paths
+    return list(filter(lambda path: (not pattern or bool(pattern.search(path))), paths))
```

Comment:
Personally I find list comprehensions easier to read:
```suggestion
    return [path for path in paths if (not pattern or bool(pattern.search(path)))]
```